crates/turborepo-errors/src/lib.rs

```diff
@@ -0,0 +1,81 @@
+use std::{
+    ops::{Deref, Range},
+    sync::Arc,
+};
+
+use biome_deserialize::{Deserializable, DeserializableValue, DeserializationDiagnostic};
+use serde::Serialize;
+
+#[derive(Debug, Default, Clone, Serialize)]
+#[serde(transparent)]
+pub struct Spanned<T> {
+    pub value: T,
+    #[serde(skip)]
+    pub range: Option<Range<usize>>,
+    #[serde(skip)]
+    pub path: Option<Arc<str>>,
+    #[serde(skip)]
+    pub text: Option<Arc<str>>,
+}
+
+impl<T: Deserializable> Deserializable for Spanned<T> {
+    fn deserialize(
+        value: &impl DeserializableValue,
+        name: &str,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        let range = value.range();
+        let value = T::deserialize(value, name, diagnostics)?;
+        Some(Spanned {
+            value,
+            range: Some(range.into()),
+            path: None,
+            text: None,
+        })
+    }
+}
+
+// We do *not* check for the range equality because that's too finicky
+// to get right in tests.
```

Comment:
Okay a `without_range` or `strip_ranges` function makes sense. For a lot of the assert_eq tests I think we should move to snapshotting with insta because it makes updating the tests a little easier versus having to manually change code. In which case we could keep the ranges